core/src/main/java/com/grappus/core/core/mvp/NavigationAction.kt

```diff
@@ -0,0 +1,22 @@
+package com.grappus.core.core.mvp
+
+/**
+ * Created by nimish@grappus.com on 15/12/2019.
+ * Description - basic NavigationActions
+ */
+
+sealed class NavigationAction {
+    data class DisplayScreen(val screenName: String, val data: Any? = null) : NavigationAction()
+
+    data class DisplayError(val errorMessage: String, val data: Any? = null) : NavigationAction()
+
+    data class Dialog(val screenName: String, val data: Any? = null)
+
+    data class SnackBar(val message: String) : NavigationAction()
+
+    data class Toast(val message: String) : NavigationAction()
+
+    data class CloseScreen(val message: String) : NavigationAction()
+
+    object AuthError : NavigationAction()
```

Comment:
Can't we just rely on `DisplayError` until any such specific need arises? Giving freedom to create custom actions may ruin the sole purpose of consistency.

Reply:
We can but this one is designed without any arguments, rest all have arguments. We can discuss it tomorrow and may be rename it to just error.